meta-facebook/gt-cc/src/platform/plat_pldm_monitor.c

```diff
@@ -0,0 +1,320 @@
+		msg.data[2] = (status >> 8) & 0xFF;
+
+		if (i2c_master_write(&msg, retry)) {
+			LOG_ERR("Clear E1S%d ISL28022 status register failed, address(0x%x) bus(0x%x)",
+				idx, msg.target_addr, msg.bus);
+			goto exec_post_exit;
+		}
+	}
+
+	if (cfg->post_sensor_read_hook) {
+		if (!cfg->post_sensor_read_hook(sensor_num, cfg->post_sensor_read_args, NULL)) {
+			LOG_ERR("SSD%d with sensor number(0x%x) post reading failed", idx,
+				cfg->num);
+		}
+	}
+
+	/* Check whether the bit is change */
+	if ((*flag & BIT(idx % 8)) ^ (is_alert << (idx % 8))) {
+		LOG_INF("Send SSD ADC alert event idx(%d) is_alert(%d)", idx, is_alert);
+
+		struct pldm_sensor_event_state_sensor_state event;
+		event.sensor_offset = PLDM_STATE_SET_OFFSET_DEVICE_STATUS;
+		event.event_state = is_alert ? PLDM_STATE_SET_OEM_DEVICE_STATUS_ALERT :
+					       PLDM_STATE_SET_OEM_DEVICE_STATUS_NORMAL;
+		event.previous_event_state = is_alert ? PLDM_STATE_SET_OEM_DEVICE_STATUS_NORMAL :
+							PLDM_STATE_SET_OEM_DEVICE_STATUS_ALERT;
+
+		if (pldm_send_platform_event(PLDM_SENSOR_EVENT, PLDM_EVENT_SENSOR_E1S_0 + idx,
+					     PLDM_STATE_SENSOR_STATE, (uint8_t *)&event,
+					     sizeof(struct pldm_sensor_event_state_sensor_state))) {
+			LOG_ERR("Send SSD%d ADC alert event log failed", idx);
+		}
+	}
+
+	/* Update the flag */
+	WRITE_BIT(*flag, idx % 8, is_alert);
+	return;
+
+exec_post_exit:
+	if (cfg->post_sensor_read_hook) {
+		if (!cfg->post_sensor_read_hook(sensor_num, cfg->post_sensor_read_args, NULL)) {
+			LOG_ERR("SSD%d with sensor number(0x%x) post reading failed", idx,
+				cfg->num);
+		}
+	}
+	return;
+}
+
+void ssd_alert_check_handler(struct k_work *work)
+{
+	struct ssd_event_check *data = CONTAINER_OF(work, struct ssd_event_check, work);
+
+	gt_power_monitor_ic_type_t type = get_power_moniter_ic_type();
+	uint8_t start_idx = (data->group == 0 ? 0 : 8);
+	uint8_t end_idx = (data->group == 0 ? 7 : 15);
+
+	for (uint8_t i = start_idx; i <= end_idx; i++) {
+		/* Pass device not present case */
+		if (gpio_get(e1s_prsnt_pin[i / 4][i % 4]))
+			continue;
+
+		if (type == POWER_IC_ISL28022) {
+			ssd_alert_check_isl28022(i, &data->flag);
+		} else if (type == POWER_IC_INA230) {
+			ssd_alert_check_ina230(i, &data->flag);
+		} else {
+			LOG_ERR("Can't send SSD%d event with unknown power IC(%d)", i, type);
+			return;
+		}
+	}
+
+	if (data->flag)
+		k_work_schedule((struct k_work_delayable *)work, K_SECONDS(5));
+}
+
+void ssd_alert_check(uint8_t group)
+{
+	if (group > 1) {
+		LOG_ERR("The argument value is invalid");
+		return;
+	}
+
+	struct ssd_event_check *data = ((group == 0) ? &ssd_event_check[0] : &ssd_event_check[1]);
+
+	if (!data->init) {
+		k_work_init_delayable(&data->work, ssd_alert_check_handler);
+		data->init = true;
+	}
+
+	data->group = group;
+
+	/* Avoid redundant workers sending */
+	if (k_work_delayable_busy_get(&data->work))
+		return;
+
+	k_work_schedule(&data->work, K_SECONDS(1));
+}
+
+void ssd_present_check()
+{
+	for (uint8_t i = 0; i < 16; i++) {
```

Comment:
Please use a #define for this instead of a magic number.

Reply:
Already update a [commit](https://github.com/facebook/OpenBIC/pull/798/commits/c23ed4e60b500713d5e8cc60f36168fa1ff736cf) modification as suggested. Please review it again, thank you.